src/liballoc/collections/btree/node.rs

```diff
@@ -508,47 +504,9 @@ impl<'a, K, V, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
 
 impl<'a, K: 'a, V: 'a, Type> NodeRef<marker::Immut<'a>, K, V, Type> {
     fn into_key_slice(self) -> &'a [K] {
```

Comment:
Maybe add a doc comment saying that it is up to the caller to ensure that this is *not* the shared root?

Reply:
For an internal function, isn't the assert obvious enough?

Reply:
It's still a big file. But I don't feel very strongly about this one.

Reply:
I would like to see this function made unsafe and a comment on it that it is unsafe to call with a shared root.

Alternatively, I think it is plausible that we can keep it safe and make the assert non-debug (realistically the one constant-comparison seems super unlikely to be the performance problem in real world code), when we're dealing with a BTreeMap that's comparing right and left anyway.